collections.js

```diff
@@ -118,8 +118,36 @@ LinkedList.prototype.find = function(value) {
     node = node.next;
   }
   return null;
+};
+
+//------------------------------------------------------------------------------
+// Simple Queue.
+
+function SimpleQueue() {
+	this.q_ = [];
 }
 
+SimpleQueue.prototype = {
+  enqueue: function(item) {
+    this.q_.push(item);
+    return this;
+  },
+
+  dequeue: function() {
+    return this.q_.shift();
```

Comment:
I initially thought that Array.shift() would be too slow, but apparently it's not that bad:
http://stackoverflow.com/questions/8423493/what-is-the-performance-of-objects-arrays-in-javascript-specifically-for-googl